src/billing-reports/consommation-report.component.tsx

```diff
@@ -0,0 +1,289 @@
+        setColumns(columnNames);
+      }
+
+      setResults(results);
+      setTotalRecords(total);
+      setPage(pageNum);
+      setPageSize(pageSize);
+      setCurrentFilters(filters);
+    } catch (error) {
+      console.error('Error fetching report:', error);
+      setErrorMessage(t('errorFetchingReport', 'Failed to load report data.'));
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleExportClick = async () => {
+    if (!currentFilters) return;
+
+    setLoading(true);
+    try {
+      const { startDate, endDate, company } = currentFilters;
+      const allResults = await fetchAllConsommationReport(formatToYMD(startDate), formatToYMD(endDate), company);
+
+      exportToExcel(columns, allResults, getValue, 'Consommations-report.xlsx');
+    } catch (error) {
+      console.error('Export failed:', error);
+      setErrorMessage(t('errorExportingExcel', 'Failed to export to Excel.'));
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const goToPage = (newPage: number) => {
+    handleSearch(currentFilters!, newPage, pageSize);
+  };
+
+  const getPageSizes = (totalItems: number, currentPageSize: number) => {
+    return [50];
+  };
+
+  const headerDisplayMap: Record<string, string> = {
+    no: t('no', 'No'),
+    date: t('date', 'Date'),
+    name: t('department', 'Department'),
+    policy_id_number: t('policyIdNumber', 'Policy ID Number'),
+    beneficiary: t('beneficiary', 'Beneficiary'),
+    insurancename: t('insuranceName', 'Insurance Name'),
+    global_amount: t('globalAmount', 'Global Amount'),
+    patientdue: t('patientDue', 'Patient Due'),
+    insurancedue: t('insuranceDue', 'Insurance Due'),
+    paid_amount: t('paidAmount', 'Paid Amount'),
+    bill_status: t('billStatus', 'Bill Status'),
+    admission_type: t('admissionType', 'Admission Type'),
+    global_bill_status: t('globalBillStatus', 'Global Bill Status'),
+    collectorname: t('collectorName', 'Collector Name'),
+  };
+
+  const filterRecordItems = (record: ReportRecord[]): ReportRecord[] => {
+    const filterColumns = [
+      'global_amount',
+      'patientdue',
+      'insurancedue',
+      'paid_amount',
+      'bill_status',
+      'admission_type',
+      'global_bill_status',
+      'collectorname',
+    ];
+
+    return record.filter((item) => filterColumns.includes(item.column));
+  };
+
+  return (
+    <div className={styles.container}>
+      {t('consommationReport', 'Consommation Report')}
+      <div className={styles.reportContent}>
+        <div className={styles.filterSection}>
+          <ReportFilterForm fields={['startDate', 'endDate', 'company']} onSearch={handleSearch} />
+        </div>
+      </div>
+
+      {loading && <p>{t('loading', 'Loading...')}</p>}
+      {errorMessage && <p className={styles.error}>{errorMessage}</p>}
+      {!loading && !errorMessage && results.length === 0 && <p>{t('noResults', 'No results found.')}</p>}
```

Comment:
Replace the simple `<p>` with a Carbon InlineNotification or an OpenMRS-style empty state, the current 'No results found' wouldn't be so helpful to the user